surrealdb.ts

```diff
@@ -0,0 +1,67 @@
+// surrealDB adapter for telegraf session
+
+import * as Surreal from "surrealdb.js";
+import { SessionStore } from "./types";
+import {defaults} from "./defaults";
+
+
+type TSurrealRecord = {
+	id: string;
+	value: any;
+}
+interface NewClientOpts {
+	url?: string;
+	auth?: Surreal.RootAuth;
+	namespace?: string;
+	database?: string;
+	onInitError?: (err: unknown) => void;
+}
+
+interface ExistingClientOpts {
+	client: Surreal.default;
+}
+
+export function SurrealDB<Session>(opts: NewClientOpts): SessionStore<Session>;
+export function SurrealDB<Session>(opts: ExistingClientOpts): SessionStore<Session>;
+export function SurrealDB<Session>(opts: NewClientOpts | ExistingClientOpts): SessionStore<Session> {
+	interface SessionDoc {
+		key: string;
+		session: Session;
+	}
+
+	let client: Surreal.default;
+
+	if ("client" in opts) client = opts.client;
+	else {
+		client = new Surreal.default(`${opts.url}/rpc`);
+		client.signin(opts.auth!)
+			.then(async () => {
+				await client.use(opts.namespace!, opts.database!);
```

Comment:
Nitpick: unnecessary async await:

```suggestion
.then(() => client.use(opts.namespace, opts.database))
```